components/widgets/menu/menu_wrapper.jsx

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import MenuWrapperAnimation from './menu_wrapper_animation.jsx';
+
+export default class MenuWrapper extends React.PureComponent {
```

Comment:
I'm not really a fan of the `FooWrapper` pattern. I feel that it makes the correct usage of the components less simple and more opaque. What is the reason that this can't just be hidden inside the `Menu` component? Looks like it relates to this component wrapping both the Menu *and* the UI element that triggers the show/hide of the Menu, but it would be good to hear directly your thinking behind this way of doing it?

Reply:
To have enough flexibility we need to have a way to pass the "clickable" thing to show the menu, the MenuWrapper is just one of the options, other would be a render prop, for example, or passing that as a "Node" property, I like how works now, but I understand that other approaches are more easy to do it right at first shot.

Reply:
OK, yeah makes sense. I don't like having a prop for the button either, so I guess this is the way to go :)

Perhaps there's a better name than `Wrapper` though. 0/5 and not blocking this PR, but I'll add it to the list of "Style Guide things we need to think about".